File: app/src/main/java/de/bitb/astroskop/model/converter/RulerConverter.java
package de.bitb.astroskop.model.converter;

import de.bitb.astroskop.model.enums.House;
import de.bitb.astroskop.model.enums.Ruler;
import io.objectbox.converter.PropertyConverter;

public class RulerConverter implements PropertyConverter<Ruler, Integer> {

    @Override
    public Ruler convertToEntityProperty(Integer databaseValue) {
        for (Ruler ruler : Ruler.values()) {
            if (ruler.getId().equals(databaseValue)) {
                return ruler;
            }
        }
        return null;
    }

    @Override
    public Integer convertToDatabaseValue(Ruler ruler) {
        return ruler.getId();
    }

}

File: app/src/main/java/de/bitb/astroskop/model/converter/RulerConverter.kt
package de.bitb.astroskop.model.converter

import de.bitb.astroskop.model.enums.Ruler
import io.objectbox.converter.PropertyConverter

class RulerConverter : PropertyConverter<Ruler, Int> {

    override fun convertToEntityProperty(databaseValue: Int?): Ruler? {
        for (ruler in Ruler.values()) {
            if (ruler.id == databaseValue) {
                return ruler
            }
        }
        return null
    }

    override fun convertToDatabaseValue(ruler: Ruler): Int? {
        return ruler.id
    }

}